### PR TITLE
fix: normalize positional-arg path separators unconditionally in cli.Execute

### LINKED ISSUE #80: fix: Windows archguard check <path> from repo root doesn't normalize path separators, breaking baseline suppression
## Problem

`cli.Execute`'s positional-arg path rewrite only normalizes separators (`filepath.ToSlash`) inside the `cwd != repoRoot` branch:

```go
if !strings.EqualFold(cwd, repoRoot) {
    for i := 2; i < len(os.Args); i++ {
        ...
        relPath = filepath.ToSlash(relPath)
        os.Args[i] = relPath
        ...
    }
}
```

When `cwd == repoRoot` -- arguably the most common invocation (`cd` into the repo, run `archguard check internal\analysis\engine.go`) -- this block is skipped entirely, so on Windows the path stored/looked-up downstream keeps backslashes.

## Impact

Surfaced while verifying PR #77 (baseline mode): `Baseline.IsSuppressed`/`Add` key on `(ADR ID, file path)`. If the file path recorded during `--update-baseline` used forward slashes (e.g. from `git`-sourced file lists) but a later `check` run's path comes from a user-typed Windows-style arg, the lookup silently misses -- baseline suppression breaks with no error, just a violation that "should" be baselined resurfacing as new drift.

This is a pre-existing bug in `internal/cli/cli.go`, not introduced by #77 -- #77 just adds a new consumer (baseline lookup) newly exposed to it.

## Suggested fix

Normalize (`filepath.ToSlash`) the rewritten positional args unconditionally, not just inside the `cwd != repoRoot` branch.

## Repro

On Windows, from the repo root:
```
archguard check internal\analysis\engine.go
```
vs. a baseline entry recorded with `internal/analysis/engine.go` (forward slashes) -- confirm whether suppression lookup succeeds.

## Acceptance criteria

- [ ] `filepath.ToSlash` is applied to every rewritten positional arg unconditionally, not only inside the `cwd != repoRoot` branch
- [ ] A test covers `cli.Execute` invoked from the repo root (`cwd == repoRoot`) with a backslash-style path arg on Windows and confirms the resulting path uses forward slashes
- [ ] A test confirms baseline suppression matches when the baseline entry was recorded with forward slashes and the `check` run is invoked with a backslash-style path from the repo root

_Filed following a red-team pass over the xhigh code review findings for PR #77._

### PR DESCRIPTION
**Pull request by Tgenz1213**
## Summary

`cli.Execute`'s positional-arg path-rewrite loop only ran when `cwd != repoRoot`, so a Windows backslash-style path arg typed from the repo root (the most common invocation) kept its backslashes instead of being normalized to forward slashes. Downstream, `internal/baseline`'s suppression lookup keys on an exact file-path string match, so a baseline entry recorded with forward slashes silently failed to match — baseline suppression broke with no error, just a violation that "should" be baselined resurfacing as new drift.

## Related issue

Closes #80

## In scope

- Extracted the path-rewrite loop into `normalizePositionalArgPaths(args []string, cwd, repoRoot string)` and call it unconditionally in `Execute`, ahead of the still-conditional `os.Chdir`.
- **Fixed during final review, not in the original issue's suggested fix**: making the call unconditional newly routes absolute-path args through `filepath.Join(cwd, arg)`, which doesn't special-case an already-absolute argument — on Linux/macOS (and Windows UNC paths) this mangled an absolute arg into a nonexistent relative path. `analysis.Engine.Run` treats a file-read failure as non-fatal (prints a warning, returns nil), so `archguard check /some/absolute/path.go` from the repo root would have silently scanned nothing and exited `0` — a false-pass in CI. Fixed with a `filepath.IsAbs` guard: a relative arg still resolves against `cwd` exactly as before; an absolute arg resolves directly against `repoRoot`. This also incidentally fixes a related pre-existing gap: an absolute in-repo path arg from *any* working directory (not just the repo root) previously normalized to an absolute path instead of the repo-relative form baseline suppression needs.

## Out of scope

- No changes to `internal/baseline` itself — its exact-string lookup behavior is unchanged; this PR fixes what gets fed into it.

## Architectural notes

No new invariant. `CLAUDE.md`'s existing `cli.Execute` footgun already describes the rewrite as applying to every positional arg; no doc update was needed since the description was already accurate about intent, just not about the code's (buggy) unconditionality.

## Follow-ups

None filed as new issues — the final review's remaining findings were genuinely Minor and non-blocking:
- An empty-string positional arg (`archguard check ""`) from the repo root now resolves to `"."`, which `resolveContentProvider` maps to a whole-repo scan rather than the loud read-error a typo previously produced. Low-severity edge case, not fixed here.
- Test coverage targets the extracted `normalizePositionalArgPaths` function directly rather than the full `Execute()` entrypoint (see QA section below) — nothing separately pins that the `Execute` call site itself stays unconditional, only that the function it calls behaves correctly. Re-wrapping that call in a conditional would reintroduce #80 with all current tests still green.

## QA / testing plan

- `go build -o archguard ./cmd/archguard` — passes.
- `go test ./...` — passes.
- `golangci-lint run --timeout=5m` — 0 issues.
- **Deliberate test-strategy deviation from the issue's literal acceptance criteria**: the issue says "a test covers `cli.Execute` invoked from the repo root..." — these tests instead target the extracted `normalizePositionalArgPaths` function directly. `Execute()` shells out to real git (`git.GetRepoRoot()`) and mutates process-global state (`os.Chdir`), so testing it directly would need much heavier scaffolding (temp git repos, cwd save/restore, no `t.Parallel`) for equivalent coverage of a regression that lives entirely in the extracted function's logic.
- Four new tests in `internal/cli/cli_test.go`, all passing:
  - `TestNormalizePositionalArgPaths_RunsEvenWhenCwdEqualsRepoRoot` — OS-agnostic (runs on Linux CI too); proves the rewrite now executes even at `cwd == repoRoot` via an uncleaned-path trick (`./sub/../file.go` → `file.go`), which can only happen if the code actually ran.
  - `TestNormalizePositionalArgPaths_HandlesAbsolutePathArg` — OS-agnostic; proves an absolute in-repo path now normalizes to the correct repo-relative form instead of being mangled.
  - `TestNormalizePositionalArgPaths_ConvertsBackslashesOnWindows` — Windows-only (`runtime.GOOS` guarded; SKIPs on Linux CI, which is expected and correct since `path/filepath`'s backslash-as-separator handling is itself GOOS-gated at compile time). Verified PASSING on this Windows dev machine, not just "presumably fine because skipped."
  - `TestNormalizePositionalArgPaths_MatchesBaselineEntryRecordedWithForwardSlashes` — Windows-only, same guard; exercises real `internal/baseline` code (`baseline.New`/`Add`/`IsSuppressed`), proving a baseline entry recorded with forward slashes now matches a backslash-typed `check` path. Verified PASSING on this Windows dev machine.

## Checklist

- [x] Title follows Conventional Commits (`feat:`, `fix:`, `docs:`, `refactor:`, `build(deps):`, etc.)
- [x] All acceptance criteria from the linked issue are met
- [ ] `go test -race -cover ./...` passes — could not run locally (no cgo/gcc on this machine, `-race` requires cgo); pending CI confirmation
- [x] `golangci-lint run --timeout=5m` is clean
- [x] `CLAUDE.md` updated if this changes build/test commands, adds or renames a top-level package, changes a cross-package interface, or adds a footgun — not needed, see Architectural notes
- [x] A new ADR added under `docs/arch/` if this embodies an architecturally-significant decision — not added; this is a bug fix restoring intended behavior, not a new invariant
- [x] Comments follow the 2-line-max, WHY-only convention

### CODE REVIEW

File: internal/cli/cli_test.go [hunk 1]
```diff
@@ -280,3 +283,67 @@ func TestBuildProvider_ClaudeAndVoyage(t *testing.T) {
...
+	cwd := repoRoot
+
+	args := []string{"archguard", "check", `internal\analysis\engine.go`}
+	normalizePositionalArgPaths(args, cwd, repoRoot)
+
+	b := baseline.New()
+	b.Add("0001", "internal/analysis/engine.go", "")
+
+	if !b.IsSuppressed("0001", args[2], "any content") {
+		t.Errorf("expected the normalized path %q to match a baseline entry recorded with forward slashes, but IsSuppressed returned false", args[2])
```
[the remark this answers is not in the document]
Comment (Tgenz1213) [reply internal/cli/cli_test.go:359]: Fixed: the test now uses a non-empty QuotedCode ("quoted violating code") with matching content, so it exercises `Baseline.IsSuppressed`'s `strings.Contains` substring check instead of the always-true empty-QuotedCode branch. Commit 5829562.

File: internal/cli/cli.go [hunk 2]
```diff
@@ -144,6 +134,24 @@ func Execute(factories ProviderFactories) (ExitCode, error) {
...
+// Must run unconditionally, not just when cwd != repoRoot -- a Windows
+// backslash-style arg typed from the repo root needs this too (see #80).
+func normalizePositionalArgPaths(args []string, cwd, repoRoot string) {
+	for i := 2; i < len(args); i++ {
+		arg := args[i]
+		if !strings.HasPrefix(arg, "-") {
+			target := arg
+			if !filepath.IsAbs(arg) {
+				target = filepath.Join(cwd, arg)
+			}
```
[the remark this answers is not in the document]
Comment (Tgenz1213) [reply internal/cli/cli.go:146]: Fixed: `normalizePositionalArgPaths` now leaves an empty positional arg untouched (added an `arg != ""` guard) instead of resolving it to ".", which `resolveContentProvider` treats as a whole-repo scan. This restores the pre-existing behavior (a mistyped empty path produces a read error via `SingleFileProvider{Path: ""}`, not a silent full scan). Added `TestNormalizePositionalArgPaths_LeavesEmptyArgUntouched` to cover it. Commit 5829562.

File: internal/cli/cli_test.go [hunk 3]
```diff
@@ -280,3 +283,67 @@ func TestBuildProvider_ClaudeAndVoyage(t *testing.T) {
...
+	if args[2] != want {
+		t.Errorf("expected an absolute in-repo path arg to normalize to the repo-relative form %q, got %q (this is the case that broke: filepath.Join(cwd, arg) mangles an already-absolute arg instead of using it directly)", want, args[2])
+	}
+}
+
+func TestNormalizePositionalArgPaths_ConvertsBackslashesOnWindows(t *testing.T) {
+	if runtime.GOOS != "windows" {
+		t.Skip("backslash-as-separator is a Windows-only path.filepath behavior")
+	}
+
```
[the remark this answers is not in the document]
Comment (Tgenz1213) [reply internal/cli/cli_test.go:332]: This is a deliberate, already-documented tradeoff, not an oversight -- pushing back rather than 'fixing' it, since there's no clean fix available.

path/filepath's backslash-as-separator handling (filepath.Join, filepath.Rel, filepath.ToSlash) is GOOS-gated at compile time, not something a test can force at runtime. On Linux, \ is just an ordinary filename character, not a separator, and filepath.ToSlash is a documented no-op when the separator is already /. So a test asserting 'backslash input becomes forward slashes' is only ever meaningful when actually built and run on Windows.

The two options to make it CI-visible would be: (1) reimplement the normalization with strings.ReplaceAll(path, "\\", "/") instead of filepath.ToSlash, which would incorrectly rewrite a legitimate Linux/macOS filename that happens to contain a literal backslash character -- overreach beyond what #80 asks for, and a real behavior change for non-Windows users; or (2) add a Windows CI runner, which is a CI-infrastructure decision well outside this PR's scope.

What IS CI-visible and does catch the actual regression class from #80: TestNormalizePositionalArgPaths_RunsEvenWhenCwdEqualsRepoRoot runs unconditionally (no GOOS guard) and proves the rewrite loop no longer gets skipped entirely at cwd == repoRoot -- the root cause of the bug -- using an OS-agnostic uncleaned-path trick (./sub/../file.go -> file.go) rather than relying on backslash semantics. The two GOOS-guarded tests exist for local verification on Windows and as executable documentation of the specific repro from the issue; I ran both on this Windows dev machine and confirmed they PASS, not just skip-and-hope.
